stdlib/public/Concurrency/Task.cpp

```diff
@@ -1747,6 +1748,21 @@ static void swift_task_startOnMainActorImpl(AsyncTask* task) {
   _swift_task_setCurrent(originalTask);
 }
 
+SWIFT_CC(swift)
+static bool swift_task_isOnMainActorImpl() {
+  // Grab the executor tracking information.
+  auto current = tasktracking::ExecutorTrackingInfo::current();
+
+  // Then if we don't have a current executor, we must be in a non-Swift
+  // Concurrency context. In that case, just check against the main thread.
+  if (!current) {
+    return Thread::onMainThread();
+  }
+
+  // Otherwise, just check if the active executor is the main executor.
+  return current->getActiveExecutor().isMainExecutor();
+}
```

Comment:
I have mixed feelings about this method because it will be wrong quite often, see how much is involved in answering this question reliably in `isCurrentExecutor`, and here as well you do tricks in Swift `_isOnMainActor` to make it actually work.

If we need this for some reason and cannot use isCurrentExecutor [in non-crashing mode ](https://github.com/swiftlang/swift/pull/76250/files#diff-03ec3b0a524a8a45afffa59620ca14f8b89487743fe498b47a5a8513579e0c7aR404)(which we should strongly prefer, rather than have many places kinda-checking-similar-thing), then we should at least name this something more specific that explains that it's not a reliable answer.